src/taskgraph/util/cached_tasks.py

```diff
@@ -7,6 +7,9 @@
 import time
 
 TARGET_CACHE_INDEX = "{cache_prefix}.cache.level-{level}.{type}.{name}.hash.{digest}"
+TARGET_PR_CACHE_INDEX = (
+    "{cache_prefix}.cache.head.{head_ref}.{type}.{name}.hash.{digest}"
```

Comment:
In theory, adding `head_ref` to this index is probably unnecessary and we could just make it a constant like "pr" or similar. But I also don't think it hurts and guarantees we only used cached tasks that were generated from the same PR. Which just feels cleaner, even if the digest basically guarantees this anyway.

Reply:
Although, if we switch to checking the PR index last (see comment below), then we'd be virtually guaranteed to never hit another PRs cache anyway. Only time it would happen is if both PRs make the exact same change to a resource file.

Reply:
My gut says that `head_ref` is probably the best way to go. `pr` has some theoretical benefits...but they seem a little unlikely and farfetched for the amount of confusion that we'd be adding. (We could always change that later though...)